### PR TITLE
Add initial metrics setup

### PR DESCRIPTION
**Pull request by KonradStaniec**
-expose metrics from covenant signer

### CODE REVIEW

File: signerservice/handlers/sign_unbonding.go [hunk 1]
```diff
@@ -66,6 +69,7 @@ func (h *Handler) SignUnbonding(request *http.Request) (*Result, *types.Error) {
 	)
 
 	if err != nil {
+		h.m.IncFailedSigningRequests()
 		// TODO Properly translate errors between layers
 		return nil, types.NewErrorWithMsg(http.StatusInternalServerError, types.InternalServiceError, err.Error())
```
Comment (gitferry) [inline signerservice/handlers/sign_unbonding.go:74]: Now sure all the errors returned from `SignUnbondingTransaction` should be counted as internal error. I think most of them are bad requests. Shall we handle them differently here?

For internal errors, maybe we can also do some logging here

**Review by gitferry — COMMENTED** [review]
One comment is that we probably should handle invalid request and internal errors differently

File: signerservice/handlers/sign_unbonding.go (hunk 1 above)
Comment (KonradStaniec) [reply signerservice/handlers/sign_unbonding.go:74]: totally forgot about this todo. I agree most of errors there will be classified as bad requests. Created issue to track this: https://github.com/babylonchain/covenant-signer/issues/29, as this is definitely not something for hot-fix which adds metrics

**Review by filippos47 — APPROVED** [review]
Thanks!
Would be nice to include the metric setup in the deployment guide, either here or on a separate PR.

File: config/metrics.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	}, nil
+}
+
+func (cfg *MetricsConfig) GetMetricsPort() int {
+	return cfg.Port
+}
+
+func DefaultMetricsConfig() *MetricsConfig {
+	return &MetricsConfig{
+		Host: "0.0.0.0",
```
Comment (filippos47) [inline config/metrics.go:43]: The default server host is `127.0.0.1`. Let's reuse the same here?

Comment (KonradStaniec) [reply config/metrics.go:43]: good point! better to not expose service by default :)

**Pull request merged**